ts/components/dialog/debug/components.tsx

```diff
@@ -81,130 +78,77 @@ export const DebugActions = () => {
             const userEd25519SecretKey = (await UserUtils.getUserED25519KeyPairBytes())
               ?.privKeyBytes;
             if (!userEd25519SecretKey) {
-              window.log.error('[debugMenu] no userEd25519SecretKey');
+              window.log.error('[debugMenu] debugLatestRelease no userEd25519SecretKey');
+              setLoadingLatestRelease(false);
               return;
             }
             setLoadingLatestRelease(true);
-            const versionNumber = await getLatestReleaseFromFileServer(userEd25519SecretKey);
-            setLoadingLatestRelease(false);
-
-            if (versionNumber) {
-              ToastUtils.pushToastInfo('debugLatestRelease', `v${versionNumber}`);
-            } else {
+            const result = await getLatestReleaseFromFileServer(userEd25519SecretKey, 'latest');
+            if (!result) {
               ToastUtils.pushToastError('debugLatestRelease', 'Failed to fetch latest release');
+              setLoadingLatestRelease(false);
+              return;
+            }
+            const [versionNumber, releaseChannel] = result;
+            if (!versionNumber) {
+              ToastUtils.pushToastError(
+                'debugLatestRelease',
+                `Failed to fetch ${releaseChannel} release`
+              );
+              setLoadingLatestRelease(false);
+              return;
             }
+            setLoadingLatestRelease(false);
+
+            ToastUtils.pushToastInfo(
+              'debugCurrentRelease',
+              `Current: v${window.versionInfo.version}`
+            );
+            ToastUtils.pushToastInfo(`debugLatestRelease`, `Available: v${versionNumber}`);
           }}
         >
           <SessionSpinner loading={loadingLatestRelease} color={'var(--text-primary-color)'} />
-          {!loadingLatestRelease ? 'Check latest release' : null}
+          {!loadingLatestRelease ? 'Check stable version' : null}
```

Comment:
same comment as below, this should be another component in the same file to clean it up a bit